src/devices_models/devices/hydro_generation.jl

```diff
@@ -409,7 +429,12 @@ function add_constraints!(
     model::DeviceModel{V, W},
     X::Type{<:PM.AbstractPowerModel},
 ) where {V <: PSY.HydroGen, W <: AbstractHydroUnitCommitment}
-    add_range_constraints!(container, T, U, devices, model, X)
+    # TODO: Confirm this is correct. Why Range in the Commitment formulation ?
```

Comment:
@sourabhdalvi please check that using range here is the correct approach.

Reply:
Seems like the commitment constraint is missing, the way we did this in v014 is adding a time series ub constraint and a commitment constraint. So, this is wrong

Reply:
We need to do another pass at checking that all the hydro stuff works.

Reply:
On second look, this looks like stale code. This method is actually calling the `add_semicontinuous_range_constraints` for hydro commitment  formulations (other than RoR)
https://github.com/NREL-SIIP/PowerSimulations.jl/blob/f2078875d3e9ab1df686d9210332e723a85582f4/src/devices_models/devices/hydro_generation.jl#L322-L332